arduino/sketch/profiles.go

```diff
@@ -0,0 +1,252 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020-2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package sketch
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/arduino/arduino-cli/arduino/utils"
+	"github.com/arduino/go-paths-helper"
+	semver "go.bug.st/relaxed-semver"
+	"gopkg.in/yaml.v2"
+)
+
+// Project represents all the profiles defined for the sketch
+type Project struct {
+	Profiles       Profiles `yaml:"profiles"`
+	DefaultProfile string   `yaml:"default_profile"`
+}
+
+// AsYaml outputs the project file as Yaml
```

Comment:
```suggestion
// AsYaml outputs the sketch project file as YAML
```
- Use standard term "sketch project file"
- Use correct case for acronym "YAML" (YAML Ain't Markup Language)